src/txn/txn_rollback_to_stable.c

```diff
@@ -1312,10 +1345,27 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
 static int
 __rollback_to_stable(WT_SESSION_IMPL *session)
 {
+    WT_CACHE *cache;
     WT_CONNECTION_IMPL *conn;
     WT_DECL_RET;
+    size_t retries;
 
     conn = S2C(session);
+    cache = conn->cache;
+
+    /*
+     * If eviction is active, toggle exclusive eviction access on and off. This will involve waiting
+     * for all active evictions to complete and give us a better chance of passing the rollback to
+     * stable check.
+     */
+#define WT_RTS_EVICT_MAX_RETRIES 5
+    for (retries = 0; retries < WT_RTS_EVICT_MAX_RETRIES && F_ISSET(cache, WT_CACHE_EVICT_ALL);
+         ++retries) {
+        /* If we're retrying, pause for a second and let eviction make some progress. */
+        if (retries != 0)
+            __wt_sleep(1, 0);
```

Comment:
This deserves a log message or something. We are spinning waiting for the system to drain resources - let's let the user know what's going on if we wait for a second or more.

Reply:
77a9a03